packages/aws-cdk-lib/aws-kinesisfirehose/lib/iceberg-destination.ts

```diff
@@ -0,0 +1,307 @@
+   * to accommodate new fields in the incoming data.
+   *
+   * Note: This feature is in preview and subject to change.
+   *
+   * @default false
+   */
+  readonly schemaEvolutionEnabled?: boolean;
+
+  /**
+   * Enable automatic table creation.
+   *
+   * When enabled, Firehose can automatically create Iceberg tables if they don't exist.
+   * Requires warehouseLocation to be specified in catalogConfiguration.
+   *
+   * Note: This feature is in preview and subject to change.
+   *
+   * @default false
+   */
+  readonly tableCreationEnabled?: boolean;
+}
+
+/**
+ * An Apache Iceberg destination for data from a Kinesis Data Firehose delivery stream.
+ */
+export class IcebergDestination implements IDestination {
+  constructor(
+    private readonly bucket: s3.IBucket,
+    private readonly props: IcebergDestinationProps,
+  ) {
+    // Validate that at least one catalog configuration is provided
+    if (!props.catalogConfiguration.catalogArn && !props.catalogConfiguration.warehouseLocation) {
+      throw new UnscopedValidationError('IcebergDestination requires at least one of catalogConfiguration.catalogArn or catalogConfiguration.warehouseLocation to be specified');
+    }
+
+    // Validate S3 backup mode - Iceberg only supports FAILED mode
+    if (this.props.s3Backup?.mode === BackupMode.ALL) {
+      throw new UnscopedValidationError('Iceberg destinations only support BackupMode.FAILED for S3 backup');
+    }
+
+    // Validate that warehouseLocation is provided when schema evolution or table creation is enabled
+    if ((props.schemaEvolutionEnabled || props.tableCreationEnabled) && !props.catalogConfiguration.warehouseLocation) {
+      throw new UnscopedValidationError('catalogConfiguration.warehouseLocation is required when schemaEvolutionEnabled or tableCreationEnabled is true');
+    }
+  }
+
+  bind(scope: Construct, _options: DestinationBindOptions): DestinationConfig {
+    // Create or use provided IAM role
+    const role = this.props.role ?? new iam.Role(scope, 'Iceberg Destination Role', {
+      assumedBy: new iam.ServicePrincipal('firehose.amazonaws.com'),
+    });
+
+    // Grant S3 permissions
+    const bucketGrant = this.bucket.grantReadWrite(role);
+
+    // Grant Glue Catalog permissions if catalogArn is specified
+    if (this.props.catalogConfiguration.catalogArn) {
+      const catalogArn = this.props.catalogConfiguration.catalogArn;
+      const actions = [
+        'glue:GetDatabase',
+        'glue:GetTable',
+        'glue:GetTableVersion',
+        'glue:GetTableVersions',
+        'glue:UpdateTable',
+      ];
+
+      // Add CreateTable permission if table creation is enabled
+      if (this.props.tableCreationEnabled) {
+        actions.push('glue:CreateTable');
+      }
+
+      role.addToPrincipalPolicy(new iam.PolicyStatement({
+        actions,
+        resources: [
+          catalogArn,
+          `${catalogArn}/database/*`,
+          `${catalogArn}/table/*`,
```

Comment:
Given that a catalog can be shared with multiple systems in a region, I feel table/* and database/* is a bit too permissive. Can we narrow down them using destinationTableConfigurations?